crates/utils/src/http_client.rs

```diff
@@ -0,0 +1,845 @@
+        let mock_server = httpmock::MockServer::start();
+
+        let mock = mock_server.mock(|when, then| {
+            when.method("GET").path("/success");
+
+            then.status(201)
+                .header("content-type", "application/json")
+                .header("x-custom", "test-value")
+                .body(r#"{"message": "created"}"#);
+        });
+
+        let client = HttpClient::builder(&mock_server.base_url()).build().unwrap();
+
+        let response = client.request().method(Method::GET).path("/success").send().await.unwrap();
+
+        assert_eq!(response.status(), 201);
+        assert_eq!(response.headers().get("content-type").unwrap(), "application/json");
+        assert_eq!(response.headers().get("x-custom").unwrap(), "test-value");
+        mock.assert();
+    }
+
+    /// Validates error response handling:
+    /// - Different HTTP error codes
+    /// - Timeout scenarios
+    /// - Network errors
+    /// - Malformed responses
+    #[tokio::test]
+    async fn test_mock_error_responses() {
+        let mock_server = httpmock::MockServer::start();
+
+        // Test 404 Not Found
+        let mock_404 = mock_server.mock(|when, then| {
+            when.method("GET").path("/not-found");
+
+            then.status(404).header("content-type", "application/json").body(r#"{"error": "not found"}"#);
+        });
+
+        // Test 500 Internal Server Error
+        let mock_500 = mock_server.mock(|when, then| {
+            when.method("GET").path("/server-error");
+
+            then.status(500).header("content-type", "application/json").body(r#"{"error": "internal server error"}"#);
+        });
+
+        // Test malformed response
+        let mock_malformed = mock_server.mock(|when, then| {
+            when.method("GET").path("/malformed");
+
+            then.status(200).header("content-type", "application/json").body("invalid json");
+        });
+
+        let client = HttpClient::builder(&mock_server.base_url()).build().unwrap();
+
+        // Test 404 response
+        let response = client.request().method(Method::GET).path("/not-found").send().await.unwrap();
+        assert_eq!(response.status(), 404);
+        mock_404.assert();
+
+        // Test 500 response
+        let response = client.request().method(Method::GET).path("/server-error").send().await.unwrap();
+        assert_eq!(response.status(), 500);
+        mock_500.assert();
+
+        // Test malformed response
+        let response = client.request().method(Method::GET).path("/malformed").send().await.unwrap();
+        assert_eq!(response.status(), 200);
+        let text = response.text().await.unwrap();
+        assert_eq!(text, "invalid json");
+        mock_malformed.assert();
+    }
+
+    /// # Error Handling Tests
+
+    /// Tests TLS certificate and identity handling:
+    /// - Valid certificate addition
+    /// - Invalid certificate handling
+    /// - Identity verification
+    #[test]
+    fn test_certificate_handling() {
```

Comment:
technically this test could be used above, no need for mock server here

Reply:
test moved and mock server removed as well